File: java/src/test/java/com/iteso/nintendo/pokemons/CharmanderTest.java
package com.iteso.nintendo.pokemons;

import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.assertEquals;

/**
 * Created by rvillalobos on 2/24/18.
 */
public class CharmanderTest {
    Charmander charmander;

    @Before
    public void setUp(){
        charmander = new Charmander();
    }

    /** Main attack test. */
    @Test
    public void mainAttackTest() {
        assertEquals("Attacking opponent with Small Fire causing a damage of 3", charmander.mainAttack());
    }

    /** Second attack test. */
    @Test
    public void secondAttackTest() {
        assertEquals("Attacking opponent with Big Fire causing a damage of 17", charmander.secondAttack());
    }

    /** Defense test */
    @Test
    public void defendTest() {
        assertEquals("Defending attack, damage caused is 0 new HP is 77", charmander.defend(0));
    }

    /** Name test */
    @Test
    public void getNameTest() {
        assertEquals("Charmander", charmander.getName());
    }

    /** Evolve test */
    @Test
    public void evolveTest() {
        assertEquals(null, charmander.evolve());
    }

    /** Type test */
    @Test
    public void typeTest() {
        assertEquals("fire", charmander.getType());
    }

    /** Evolvution test */
    @Test
    public void hasEvolutionTest() {
        assertEquals(true, charmander.hasEvolution());
    }

    /** Main Damage test */
    @Test
    public void getMainAttackDamageTest() {
        assertEquals(3, charmander.getMainAttackDamage());
    }

    /** Second Damage test */
    @Test
    public void getSecondAttackDamageTest() {
        assertEquals(17, charmander.getSecondAttackDamage());
    }

    /** Hit points test */
    @Test
    public void getHitPointsTest() {
        assertEquals(77, charmander.getHitPoints());
    }

    /** Defense multiplier test */
    @Test
    public void getDefenseMultiplierTest() {
        assertEquals(0.3, charmander.getDefenseMultiplier(), 0.0);
    }
}

File: python/src/unittest/python/CharmanderTest.py
import unittest
from Charmander import *


class CharmanderTest(unittest.TestCase):
    def setUp(self):
        self.charmander = Charmander()

    def mainAttackTest(self):
        self.assertEquals(
            "Attacking opponent with Small Fire causing a damage of 3",
            self.charmander.mainAttack())

    def secondAttackTest(self):
        self.assertEquals(
            "Attacking opponent with Big Fire causing a damage of 17",
            self.charmander.secondAttack())

    def defendTest(self):
        self.assertEquals(
            "Defending attack, damage caused is 0 new HP is 77",
            self.charmander.defend(0))

    def getNameTest(self):
        self.assertEquals("Charmander", self.charmander.getName())

    def evolveTest(self):
        self.assertEquals(None, self.charmander.evolve())

    def typeTest(self):
        self.assertEquals("fire", self.charmander.getType())

    def hasEvolutionTest(self):
        self.assertEquals(True, self.charmander.isHasEvolution())


    def getMainAttackDamageTest(self):
        self.assertEquals(3, self.charmander.getMainAttackDamage())

    def getSecondAttackDamageTest(self):
        self.assertEquals(17, self.charmander.getSecondAttackDamage())

    def getDefenseMultiplierTest(self):
        self.assertEquals(0.3, self.charmander.getDefenseMultiplier())

    def getHitPointsTest(self):
        self.assertEquals(77, self.charmander.getHitPoints())